src/Bundler.js

```diff
@@ -448,12 +449,31 @@ class Bundler extends EventEmitter {
     return Server.middleware(this);
   }
 
-  serve(port = 1234) {
-    this.logger.persistent(
-      'Server running at ' + this.logger.chalk.cyan(`http://localhost:${port}`)
-    );
+  async serve(port = 1234) {
     this.bundle();
-    return Server.serve(this, port);
+    let server = await Server.serve(this, port);
+
+    server.once('error', err => {
+      this.logger.error(
+        new Error(customErrors.serverErrors(err, server.address().port))
+      );
+    });
+
+    server.once('listening', connection => {
+      let addon =
+        server.address().port !== port
+          ? `- ${this.logger.chalk.red(
+              `configured port ${port} could not be used.`
+            )}`
+          : '';
+      this.logger.persistent(
+        `Server running at ${this.logger.chalk.cyan(
+          `http://localhost:${server.address().port}`
+        )} ${addon}\n`
+      );
+    });
+
+    return server;
```

Comment:
let's put this code in `Server.js` instead of here. Trying to avoid adding too much more to the `Bundler` :)

Reply:
fixed in aaa5308c2dac6ddb3b637d0c9bbaff8f12ebca97